dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java

```diff
@@ -269,6 +277,107 @@ public void findAllTest() throws Exception {
         ;
     }
 
+    @Test
+    public void findAllNoDuplicatesOnMultipleCommunityTitlesTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        List<String> titles = Arrays.asList("First title", "Second title", "Third title", "Fourth title");
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName(titles.get(0))
+                .withTitle(titles.get(1))
+                .withTitle(titles.get(2))
+                .withTitle(titles.get(3))
+                .build();
+
+        Community child1 = CommunityBuilder.createSubCommunity(context, parentCommunity)
+                                           .withName("Sub Community")
+                                           .build();
+
+
+        getClient().perform(get("/api/core/communities").param("size", "2"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$._embedded.communities", Matchers.containsInAnyOrder(
+                        CommunityMatcher.matchCommunityEntryMultipleTitles(titles, parentCommunity.getID(),
+                                parentCommunity.getHandle()),
+                        CommunityMatcher.matchCommunityEntry(child1.getID(), child1.getHandle())
+                )))
+                .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/core/communities")))
+                .andExpect(jsonPath("$.page.totalElements", is(2)))
+                .andExpect(jsonPath("$.page.totalPages", is(1)));
+    }
+
+    @Test
+    public void findAllNoDuplicatesOnMultipleCommunityTitlesPaginationTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        List<String> titles = Arrays.asList("First title", "Second title", "Third title", "Fourth title");
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                                          .withName(titles.get(0))
+                                          .withTitle(titles.get(1))
+                                          .withTitle(titles.get(2))
+                                          .withTitle(titles.get(3))
+                                          .build();
+
```

Comment:
to test what was described here 
https://github.com/DSpace/DSpace/pull/2486/files#r328349364

please add two other communities one with a title and one without title and make the call asking for a page size = 2.

You should get 3 as total element, 2 in the first page and 1 in the second page (it would be better to check also the retrieval of the second page so that we will stress the offset parameter)